orangecontrib/text/tests/test_twitter.py

```diff
@@ -0,0 +1,192 @@
+
+        def on_finish():
+            self.assertEqual(self.checker, 3)
+            self.checker += 1
+
+        api = twitter.TwitterAPI(self.credentials, on_start=on_start,
+                                 on_progress=on_progress, on_finish=on_finish)
+        api.search(word_list=['hello'], max_tweets=2, lang='en')
+        api.join()
+        self.assertEqual(self.checker, 4)
+
+        if NO_CREDENTIALS:
+            self.skipTest(CREDENTIALS_MSG)
+
+        def on_error(error):
+            self.assertEqual(self.checker, 4)
+            self.checker += 1
+
+        api = twitter.TwitterAPI(twitter.Credentials('bad', 'credentials'),
+                                 on_error=on_error)
+
+        api.search(word_list=['hello'], max_tweets=2, lang='en')
+        api.join()
+        self.assertEqual(self.checker, 5)
+
+    @unittest.skipIf(NO_CREDENTIALS, CREDENTIALS_MSG)
+    def test_search_disconnect(self):
+        self.api.search(word_list=['hello'], max_tweets=50, lang='en')
+        self.api.disconnect()
+        self.api.join()
+        self.assertLess(len(self.api.container), 20)
+
+    def test_create_corpus(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        corpus = self.api.create_corpus()
+        self.assertIsInstance(corpus, Corpus)
+        self.assertEqual(len(corpus), 5)
+
+    def test_crate_corpus_attr_selection(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        attributes = ['text', 'created_at', 'author_id']
+        corpus = self.api.create_corpus(included_attributes=attributes)
+        domain_attributes = [attr.name for attr in chain(corpus.domain.attributes, corpus.domain.metas)]
+        self.assertEqual(len(domain_attributes), 3)
+        for attr in attributes:
+            self.assertIn(attr, domain_attributes)
+
+    def test_clear(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        self.assertEqual(len(self.api.container), 5)
+
+        self.api.reset()
+        self.assertEqual(len(self.api.container), 0)
+
+    @unittest.skipIf(NO_CREDENTIALS, CREDENTIALS_MSG)
+    def test_running(self):
+        self.assertFalse(self.api.running)
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.assertTrue(self.api.running)
+        self.api.disconnect()
+        self.assertFalse(self.api.running)
+
+    def test_report(self):
+        self.api.search(word_list=['hello'], max_tweets=5)
+        self.api.join()
+        self.assertEqual(len(self.api.history), 1)
+        self.assertIsNotNone(self.api.task.report())
+
+    def test_geo_util(self):
+        point = twitter.coordinates_geoJSON({})
+        self.assertIsNone(point[0])
+        self.assertIsNone(point[1])
+
+        point = twitter.coordinates_geoJSON({'coordinates': [10, 10]})
+        self.assertEqual(point[0], 10)
+        self.assertEqual(point[1], 10)
+
+    def test_build_query(self):
```

Comment:
Wouldn't it be better to use `self.assestEqual(query, '<CORRECT_QUERY>')`? Current tests do not assure that the query is correct. They just check that e.g. `2016` is in query which would also pass if there would be `201620162016` in query.